internal/pkg/cli/env_init.go

```diff
@@ -206,9 +206,25 @@ func (o *initEnvOpts) Validate() error {
 			return err
 		}
 	}
+
 	if o.AppName() == "" {
-		return fmt.Errorf("no application found: run %s or %s into your workspace please", color.HighlightCode("app init"), color.HighlightCode("cd"))
+		existingApps, _ := o.store.ListApplications()
+		if len(existingApps) == 0 {
+			o.askAppName(fmtAppInitNamePrompt)
+		} else {
+			useExistingApp, err := o.prompt.Confirm(
+			"Would you like to use one of your existing applications?", "", prompt.WithTrueDefault(), prompt.WithFinalMessage("Use existing application:"))
+			if err != nil {
+				return fmt.Errorf("prompt to confirm using existing application: %w", err)
+			}
+			if useExistingApp {
```

Comment:
Can we use the Application() method on the configSelector method? You can see an example of this in `internal/pkg/cli/app_show.go` with how we initialize the `appSelector` object on [line 46](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/cli/app_show.go#L46) and call the `Application()` method on [line 130](https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/cli/app_show.go#L130).

This method handles all the short-circuiting logic if there's only one app in an account :)